File: sandbox/exercism/c/2020/resistor_color.c
/* Resistor Color from https://exercism.io */

#include <stdlib.h>
#include "resistor_color.h"

int color_code(resistor_band_t color)
{
  return color;
}

resistor_band_t *colors()
{
  resistor_band_t *codes = malloc(sizeof(resistor_band_t) * WHITE);

  for (resistor_band_t i = BLACK; i <= WHITE; i++) {
    codes[i] = i;
  }

  return codes;
}

File: sandbox/exercism/go/2025/resistor_color.go
// https://exercism.org/tracks/go/exercises/resistor-color

package resistorcolor

func Colors() []string {
	return []string{"black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"}
}

func ColorCode(color string) int {
	var code int

	for i, c := range Colors() {
		if c == color {
			code = i
			break
		}
	}

	return code
}
